corehq/apps/app_manager/static/app_manager/js/forms/form_workflow.js

```diff
@@ -192,6 +194,26 @@ hqDefine('app_manager/js/forms/form_workflow', function () {
             return (!self.autoLink() || self.manualDatums());
         });
 
+        if (self.advancedMode) {
+            self.advancedModeDatumsElement = uiElementKeyValueList.new(
+                String(Math.random()).slice(2),  // random id
+                gettext("Manual Linking Datums"),
+                gettext("Set datums required to navigate to the selected form or menu"),
+                {key: gettext("Datum ID"), value: gettext("XPath Expression")}  // placeholders
+            );
+            let datumsDict = {};
+            _.each(datums, function (datum) {
+                datumsDict[datum.name] = datum.xpath;
+            });
+            self.advancedModeDatumsElement.val(datumsDict);
+            self.advancedModeDatumsElement.on("change", function () {
+                self.serializedDatums(JSON.stringify(_.map(this.val(), function (xpath, name) {
```

Comment:
Just making sure I understand this correctly. If we are not in advanced mode `self.serializedDatums` will be used by the `formId` element. But you can only be in "normal" or "advanced". So there is no conflict? Is that correct?

Reply:
That's right - `serializedDatums` will be controlled by one or the other.  I do wish this had further separation though, like some sort of composable element that treats the variable stuff as a widget where you can swap between one or the other.  Though I think if I put much more work into this UI I'd probably drop this widget entirely and work on combining the two UIs (maybe, I don't know - I don't trust the old UI, and that might be worse, for instance if the displayed case type is wrong).

Reply:
I feel like having a follow up story/pr for cleaning up the config ui would be reasonable